### ISSUE TITLE
createCustomToken fails error: Using integers for registered date claims is deprecated, please use DateTimeImmutable objects instead

### ISSUE THREAD

**Issue by kevin-boom** [kreait/firebase-php#504]
**Describe the bug**

This is the error: 

` "\/www\/vendor\/lcobucci\/jwt\/src\/Builder.php(151): Using integers for registered date claims is deprecated, please use DateTimeImmutable objects instead."`

Version of lcobucci/jwt v3.4

**To Reproduce**
I'm using v5.11

composer automatically installs lcobucci/jwt v3.4 as a dependency of kreait/firebase-php

```
$auth = $myFirebaseInstance->createAuth(); //successfully created firebase $auth object

$customToken = $auth->createCustomToken($uid, ["email" =>"foo@bar.com", "myBool"=>true, "disabled"=>false ]); //fails with deprecation error 

```

**Expected behavior**
Expected to receive $customToken

**Screenshots**

**Environment (please complete the following information):**
 - PHP version: 7.3.14
 - Firebase SDK Version: 5.11

**Additional context**

**Comment by kevin-boom** [comment]
Fixed this issue by manually specifying Icobucci/jwt v3.3.3

Composer automatically installs 3.4 which causes this error

**Comment by jeromegamez** [comment]
Thanks for reporting this, I will replace the deprecated calls with https://github.com/kreait/firebase-php/pull/496 - but I'm wondering why this causes a fail - it's just a deprecation an should not interrupt the application flow 🤔

**Comment by jeromegamez** [comment]
In the meantime please see https://github.com/lcobucci/jwt/issues/550 - there are some approaches in the thread on how to handle the deprecation messages that explain them better than I could :)

**Comment by sdwru** [comment]
So what is the solution in the meantime?  Sounds like downgrading to "lcobucci/jwt": "^3.3.0" but if we are installing this package from composer we can't do that.  Only the authors of this package can do that unless we link directly to this github repo which defeats the purpose of using packagist.

**Comment by lcobucci** [comment]
@jeromegamez sorry to hear this is also affecting your work here.

Feel free to ping me on the PR that tackles the API upgrade (the guide I wrote should provide enough info :crossed_fingers:).

https://github.com/laravel/framework/pull/35381 will hopefully get merged soon and should help Laravel users with this thing too.

**Comment by sdwru** [comment]
> Fixed this issue by manually specifying Icobucci/jwt v3.3.3
> 
> Composer automatically installs 3.4 based off kreait/firebase's dependencies which causes this error

Where did you do that.  Inside `/vendor/kreait/firebase-php/composer.json` or in your root `composer.json`?

**Comment by lcobucci** [comment]
@sdwru pro-tip: avoid manual edits on `composer.json`. Using `composer require` is less error-prone and allows you to only affect the dependencies you want.

In this case, you'd run `composer require lcobucci/jwt:3.3.3 --update-with-all-dependencies`.

**Comment by jeromegamez** [comment]
@lcobucci Thanks for stopping by 🤩 and for the ping-invitation - I'm tackling this in additional commits to Feature-PRs here and the underlying Firebase tokens library that I planned to release soon anyway, and wouldn't want to bother you even more, you certainly have enough on your plate already.

@sdwru Please do it in the root `composer.json`, you should never make manual changes in the `vendor` directory as they are easily overwritten or lead to conflicts by subsequent `composer update`s.

I expect the new releases to be ready within the next 24 hours.

**Comment by jeromegamez** [comment]
Fixed with 6207e7e7c221f536a5ac2c125832a6f8d37c0fe1 and https://github.com/kreait/firebase-tokens-php/commit/d9ac3e70f366c0b06d41405f0962eeb3e2ff3a51 - released with [5.12.0](https://github.com/kreait/firebase-php/releases/tag/5.12.0) 🚀
